tfjs-backend-webgl/src/gpgpu_math.ts

```diff
@@ -409,10 +410,10 @@ export function makeShaderKey(
       // |rank34| is used in getSampler3D/getSampler4D.
       // |isTexShapeGreaterThanOne| are used in
       // getSampler[Scalar|1D|2D]/getOutput1DCoords.
-      keyInputs += `${xRank}_${isInOutTexShapeEqual}_${useSqueezeShape}_${
-          uniformShape.length}_${isScalar}_${broadcastDims}_${
-          isLogicalShapTexShapeEqual}_${rank1}_${rank2}_${rank34}_${
-          isTexShapeGreaterThanOne}_${hasOffset}`;
+      keyInputs += `${xRank}_${isInOutTexShapeEqual}_${
+          useSqueezeShape ? keptDims : ''}_${uniformShape.length}_${isScalar}_${
```

Comment:
Some failures are caused by missing `keptDims` in shader key.
Previously, we couldn't distinguish
```
float getA(int row, int col, int depth)
{
  return getA(col, depth);
}
```
vs
```
float getA(int row, int col, int depth)
{
  return getA(row, depth);
}
```